### PR TITLE
Add Cloud Run  sample

### PR DESCRIPTION
**Pull request by averikitsch**
Fixes #issue

> It's a good idea to open an issue first for discussion.

- [ ] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] API's need to be enabled to test (tell us)
- [ ] Environment Variables need to be set (ask us to set them)
- [ ] **Tests** pass:   `mvn clean verify` **required**
- [ ] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: run/filesystem/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+WORKDIR /app
+COPY pom.xml .
+COPY src ./src
+
+# Build a release artifact.
+RUN mvn package -DskipTests
+
+# Use AdoptOpenJDK for base image.
+# https://docs.docker.com/develop/develop-images/multistage-build/#use-multi-stage-builds
+FROM adoptopenjdk/openjdk11:debian
```
Comment (lesv) [inline run/filesystem/Dockerfile:31]: adoptopenjdk has changed their name, use `eclipse-temurin` instead.  See https://hub.docker.com/_/eclipse-temurin for tag info.

File: run/filesystem/src/test/java/com/example/filesystem/FilesystemApplicationIT.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        "alpha",
+        "run",
+        "deploy",
+        service,
+        "--source=.",
+        "--region=us-central1",
+        "--no-allow-unauthenticated",
+        String.format("--vpc-connector=%s", connector),
+        "--execution-environment=gen2",
+        String.format("--update-env-vars=IP_ADDRESS=%s,FILE_SHARE_NAME=vol1", ipAddress));
```
Comment (lesv) [inline run/filesystem/src/test/java/com/example/filesystem/FilesystemApplicationIT.java:70]: Wow - that doesn't look robust.  Ok if it's what you need.

File: run/filesystem/src/test/java/com/example/filesystem/FilesystemApplicationTests.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  private static Map<String, String> getModifiableEnvironment() throws Exception {
+    Class pe = Class.forName("java.lang.ProcessEnvironment");
+    Method getenv = pe.getDeclaredMethod("getenv");
+    getenv.setAccessible(true);
+    Object unmodifiableEnvironment = getenv.invoke(null);
+    Class map = Class.forName("java.util.Collections$UnmodifiableMap");
+    Field m = map.getDeclaredField("m");
+    m.setAccessible(true);
+    return (Map) m.get(unmodifiableEnvironment);
+  }
```
Comment (lesv) [inline run/filesystem/src/test/java/com/example/filesystem/FilesystemApplicationTests.java:105]: You might wish to explain why you are doing this.  Are you just trying to see if I'm reading?

**Review by lesv — APPROVED** [review]
Some things in here look fragile or a bit sketchy, you might wish to add comments as to why you are doing things.

File: run/filesystem/src/test/java/com/example/filesystem/FilesystemApplicationTests.java (hunk 3 above)
Comment (averikitsch) [reply run/filesystem/src/test/java/com/example/filesystem/FilesystemApplicationTests.java:105]: This is sketchy but I need a way to set an environment variable for my test.

File: run/filesystem/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START cloudrun_fs_dockerfile]
+
+# Use the official maven/Java 11 image to create a build artifact.
+# https://hub.docker.com/_/maven
+FROM maven:3.8-jdk-11 as builder
```
[the remark this answers is not in the document]
Comment (averikitsch) [reply run/filesystem/Dockerfile]: done

File: run/filesystem/Dockerfile (hunk 1 above)
Comment (averikitsch) [reply run/filesystem/Dockerfile]: done

**Pull request merged**